transactor/src/main/scala/io/mediachain/copycat/Client.scala

```diff
@@ -121,21 +122,37 @@ class Client(client: CopycatClient) extends JournalClient {
     stateListeners.foreach(_.onStateChange(Disconnected))
   }
   
+  private def recover() {
+    val cf = client.recover()
+    recoveryExecutor.submit(new Runnable {
+      def run {
+        try {
+          cf.join()
+          logger.info("Copycat session recovered")
+        } catch {
+          case e: InterruptedException => ()
```

Comment:
If you're trying to catch a an `InterruptedException` inside the future that you've joined to, you need to catch `CompletionException` and look at underlying with `getCause`